File: VB/DxSample/Scheduler/CustomAppointment.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraScheduler

Namespace DxSample.Scheduler
	<Serializable> _
	Public Class CustomAppointment
		Inherits Appointment
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal type As AppointmentType)
			MyBase.New(type)
		End Sub
		Public Sub New(ByVal start As DateTime, ByVal [end] As DateTime)
			MyBase.New(start, [end])
		End Sub
		Public Sub New(ByVal start As DateTime, ByVal duration As TimeSpan)
			MyBase.New(start, duration)
		End Sub
		Public Sub New(ByVal type As AppointmentType, ByVal start As DateTime, ByVal [end] As DateTime)
			MyBase.New(type, start, [end])
		End Sub
		Public Sub New(ByVal type As AppointmentType, ByVal start As DateTime, ByVal duration As TimeSpan)
			MyBase.New(type, start, duration)
		End Sub
		Public Sub New(ByVal start As DateTime, ByVal [end] As DateTime, ByVal subject As String)
			MyBase.New(start, [end], subject)
		End Sub
		Public Sub New(ByVal start As DateTime, ByVal duration As TimeSpan, ByVal subject As String)
			MyBase.New(start, duration, subject)
		End Sub
		Public Sub New(ByVal type As AppointmentType, ByVal start As DateTime, ByVal [end] As DateTime, ByVal subject As String)
			MyBase.New(type, start, [end], subject)
		End Sub
		Public Sub New(ByVal type As AppointmentType, ByVal start As DateTime, ByVal duration As TimeSpan, ByVal subject As String)
			MyBase.New(type, start, duration, subject)
		End Sub
		Public Sub New(ByVal type As AppointmentType, ByVal start As DateTime, ByVal duration As TimeSpan, ByVal subject As String, ByVal id As Object)
			MyBase.New(type, start, duration, subject, id)
		End Sub

		Friend Sub SetRowHandleInternal(ByVal handle As Object)
			RowHandle = handle
		End Sub

		Private privateObjectForInsert As Object
		Public Property ObjectForInsert() As Object
			Get
				Return privateObjectForInsert
			End Get
			Set(ByVal value As Object)
				privateObjectForInsert = value
			End Set
		End Property

		Protected Overrides Sub Assign(ByVal src As Appointment)
			MyBase.Assign(src)
			ObjectForInsert = (CType(src, CustomAppointment)).ObjectForInsert
		End Sub
	End Class
End Namespace

File: CS/DxSample/Scheduler/CustomAppointment.cs
using System;
using DevExpress.XtraScheduler;

namespace DxSample.Scheduler {
    [Serializable]
    public class CustomAppointment :Appointment {
        public CustomAppointment() : base() { }
        public CustomAppointment(AppointmentType type) : base(type) { }
        public CustomAppointment(DateTime start, DateTime end) : base(start, end) { }
        public CustomAppointment(DateTime start, TimeSpan duration) : base(start, duration) { }
        public CustomAppointment(AppointmentType type, DateTime start, DateTime end) : base(type, start, end) { }
        public CustomAppointment(AppointmentType type, DateTime start, TimeSpan duration) : base(type, start, duration) { }
        public CustomAppointment(DateTime start, DateTime end, string subject) : base(start, end, subject) { }
        public CustomAppointment(DateTime start, TimeSpan duration, string subject) : base(start, duration, subject) { }
        public CustomAppointment(AppointmentType type, DateTime start, DateTime end, string subject) : base(type, start, end, subject) { }
        public CustomAppointment(AppointmentType type, DateTime start, TimeSpan duration, string subject) : base(type, start, duration, subject) { }
        public CustomAppointment(AppointmentType type, DateTime start, TimeSpan duration, string subject, object id) : base(type, start, duration, subject, id) { }

        internal void SetRowHandleInternal(object handle) {
            RowHandle = handle;
        }

        public object ObjectForInsert { get; set; }

        protected override void Assign(Appointment src) {
            base.Assign(src);
            ObjectForInsert = ((CustomAppointment)src).ObjectForInsert;
        }
    }
}
